### PR TITLE
Evaluation framework for Selenium

### PR DESCRIPTION
**Pull request by dhuynh95**
I have provided the code to easily benchmark our action engine pipeline (both on the retriever and LLM) for Selenium code.

To make code clean, I had to slightly change the action engine.

@mbrunel what do you think? We can sit together if you want to review it before merge. It's not rocket science

### CODE REVIEW

File: src/lavague/defaults.py [hunk 1]
```diff
@@ -91,6 +91,48 @@ def default_get_selenium_driver() -> SeleniumDriver:
...
+        chrome_options = Options()
+        chrome_options.add_argument("--headless")  # Ensure GUI is off
+        chrome_options.add_argument("--no-sandbox")
+        chrome_options.add_argument("--window-size=1600,900")
+        chrome_options.add_experimental_option("prefs", {
+            "profile.managed_default_content_settings.javascript": 2
+        })
+        chrome_options.add_argument('--proxy-server=127.0.0.1:9999')
+
+        homedir = os.path.expanduser("~")
```
Comment (shubhamofbce) [inline src/lavague/defaults.py:113]: I don't think we need a new function for this, we can keep it in the default get driver method like this and if evaluation is true, we will add those new options.
```python
def default_get_selenium_driver(evaluation=False):
    if evaluation:
        add options
    


```

File: src/lavague/evaluator.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    return False
+
+def extract_backend_node_ids(html_content):
+    soup = BeautifulSoup(html_content, 'html.parser')
+    return set([tag['backend_node_id'] for tag in soup.find_all(attrs={"backend_node_id": True})])
+
+def id_recall(ground_truth_outer_html, retrieved_context):
+    ground_truth_ids = extract_backend_node_ids(ground_truth_outer_html)
+    context_ids = extract_backend_node_ids(retrieved_context)
+    recall = len(ground_truth_ids & context_ids) / len(ground_truth_ids)
```
Comment (shubhamofbce) [inline src/lavague/evaluator.py:84]: For some reason, when i am running locally
len(ground_truth_ids) is 0 and throwing `ZeroDivisionError: division by zero`

Comment (shubhamofbce) [inline src/lavague/evaluator.py:84]: For some reason,
len(ground_truth_ids) is zero on my local and throwing
```
ZeroDivisionError: division by zero```

Comment (dhuynh95) [reply src/lavague/evaluator.py:84]: @shubhamofbce the unique IDs might not have been injected then, have to check

**Pull request merged**